core/src/test/java/io/jenkins/pluginhealth/scoring/probes/IncrementalBuildDetectionProbeTest.java

```diff
@@ -0,0 +1,98 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.nio.file.Path;
+import java.util.Map;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class IncrementalBuildDetectionProbeTest extends AbstractProbeTest<IncrementalBuildDetectionProbe> {
+    private Plugin plugin;
+    private ProbeContext ctx;
+    private IncrementalBuildDetectionProbe probe;
+
+    @BeforeEach
+    public void init() {
+        plugin = mock(Plugin.class);
+        ctx = mock(ProbeContext.class);
+        probe = getSpy();
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            ContinuousDeliveryProbe.KEY, ProbeResult.success(ContinuousDeliveryProbe.KEY, "")
+        ));
+    }
+
+    @Override
+    IncrementalBuildDetectionProbe getSpy() {
+        return spy(IncrementalBuildDetectionProbe.class);
+    }
+
+    @Test
+    void shouldReturnASuccessfulCheckIncrementalBuildConfiguredInBothFiles() {
+        when(ctx.getScmRepository()).thenReturn(Path.of("src/test/resources/jenkinsci/test-plugin-1"));
+        assertThat(probe.apply(plugin, ctx))
+            .usingRecursiveComparison()
+            .comparingOnlyFields("id", "message", "status")
+            .isEqualTo(ProbeResult.success(IncrementalBuildDetectionProbe.KEY, "Incremental Build is configured in the plugin."));
+        verify(probe).doApply(plugin, ctx);
+    }
+
+    @Test
+    void shouldReturnASuccessfulCheckWhenIncrementalBuildIsConfiguredOnlyInExtensionsXML() {
```

Comment:
Not the expected behavior. We need both files to be present, not just one of them.